src/Security/Authentication/Cookies/src/CookieAuthenticationHandler.cs

```diff
@@ -91,7 +91,15 @@ private void CheckForRefresh(AuthenticationTicket ticket)
                 var timeElapsed = currentUtc.Subtract(issuedUtc.Value);
                 var timeRemaining = expiresUtc.Value.Subtract(currentUtc);
 
-                if (timeRemaining < timeElapsed)
+                var eventContext = new CookieSlidingExpirationContext(Context, Scheme, Options, ticket)
```

Comment:
Can we do this only if there's an event?

Reply:
We could, though it would be a departure from the existing eventing pattern where all events have a default. That was originally done to reduce implementation complexity.

We never did attempt an allocation profile of cookie auth 😁.